infra/gcp/bash/ensure-releng.sh

```diff
@@ -44,6 +44,103 @@ if [ $# = 0 ]; then
+    
+    # Check if the constraint is enabled
+    local status_file
+    status_file=$(mktemp)
+    if gcloud --project="${project}" org-policies describe \
+        iam.disableCrossProjectServiceAccountUsage > /dev/null 2>"${status_file}"; then
+        return
+    fi
+
+    # If checking the contraint was not successful, check the error
+    if ! (grep "NOT_FOUND:" "${status_file}"); then
+        echo "Error checking cross-project contraint status" >&2
+        return 1
+    fi
+
+    # Its enforced, disable it with a policy
+    echo " >> disabling cross-project impersonation enforcement"
+    local policy_file
+    policy_file=$(mktemp)
+    {
+        echo "name: projects/${project_id}/policies/iam.disableCrossProjectServiceAccountUsage"
+        echo "spec:"
+        echo "  rules:"
+        echo "  - enforce: false" 
+    } > "${policy_file}"
+    gcloud org-policies set-policy "${policy_file}" || return 1
+}
+
+# Ensure the signer service accounts exist and the
+# required accounts have access to them as token creators
+function ensure_signer_service_accounts() {
+    if [ $# != 1 ] || [ -z "$1" ]; then
+        echo "ensure_signer_service_accounts requires a project name" >&2
+        return 1
+    fi
+    
+    local project="${1}"
+
+    # Service account for signing artifacts during staging
+    color 6 "Ensuring staging signing account"
+    ensure_service_account \
+        "${project}" \
+        "${STAGING_SIGNER_SVCACCT}" \
+        "krel staging signing account"
+
+    # Service account for signing artifacts during staging
+    color 6 "Ensuring main Kubernetes org signing account"
+    ensure_service_account \
+        "${project}" \
+        "${K8s_ORG_SIGNER_SVCACCT}" \
+        "Kubernetes signer account"
+        
+    # The cloud build account needs to be able to produce OIDC identity
+    # tokens with the staging signer identity
+    staging_sign_account="$(svc_acct_email "${project}" "${STAGING_SIGNER_SVCACCT}")"
+    ensure_serviceaccount_role_binding \
+        "${staging_sign_account}" \
+        "serviceAccount:${RELEASE_PROCESS_CLOUDBUILD_SVCACCT}" \
+        "roles/iam.serviceAccountTokenCreator"
+
+    # The image promoter accounts that handle image and file promotion need
+    # access to the main Kubernetes signing account to produce OIDC tokens
+    # with its identity. 
+    prod_sign_account="$(svc_acct_email "${project}" "${K8s_ORG_SIGNER_SVCACCT}")"
+    promoter_file_account="$(svc_acct_email "${PROMOTER_PROJECT}" "${FILE_PROMOTER_SVCACCT}")"
+    promoter_image_account="$(svc_acct_email "${PROMOTER_PROJECT}" "${IMAGE_PROMOTER_SVCACCT}")"
+    
+    ensure_serviceaccount_role_binding \
+        "${prod_sign_account}" \
+        "serviceAccount:${promoter_file_account}" \
+        "roles/iam.serviceAccountTokenCreator"
+    
+    ensure_serviceaccount_role_binding \
+        "${prod_sign_account}" \
+        "serviceAccount:${promoter_image_account}" \
+        "roles/iam.serviceAccountTokenCreator"
+}
```

Comment:
Looks...like a duplicate ?

Reply:
Nope, files and images are promoted by different identities:

`k8s-infra-promoter` vs `k8s-infra-gcr-promoter`